sn_node/src/event.rs

```diff
@@ -77,6 +78,43 @@ pub enum NodeEvent {
     },
 }
 
+impl std::fmt::Debug for NodeEvent {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NodeEvent::ConnectedToNetwork => write!(f, "ConnectedToNetwork"),
+            NodeEvent::ChunkStored(address) => write!(f, "ChunkStored({:?})", address),
+            NodeEvent::RegisterCreated(address) => write!(f, "RegisterCreated({:?})", address),
+            NodeEvent::RegisterEdited(address) => write!(f, "RegisterEdited({:?})", address),
+            NodeEvent::SpendStored(pubkey) => write!(f, "SpendStored({:?})", pubkey),
+            NodeEvent::ChannelClosed => write!(f, "ChannelClosed"),
+            NodeEvent::BehindNat => write!(f, "BehindNat"),
+            NodeEvent::GossipsubMsg { topic, msg } => {
+                write!(f, "GossipsubMsg {{ topic: {:?}, msg: {:?} }}", topic, msg)
+            }
+            NodeEvent::TransferNotif { key, transfers } => {
+                let transfers_hashes = calculate_hashes(transfers); // Assuming a function to calculate hashes
+                write!(
+                    f,
+                    "TransferNotif {{ key: {:?}, transfers_hashes: {:?} }}",
+                    key, transfers_hashes
+                )
+            }
+        }
+    }
+}
+
+/// Function to calculate hashes of each transfer
+fn calculate_hashes(transfers: &[Transfer]) -> Vec<u64> {
+    transfers
+        .iter()
+        .map(|transfer| {
+            let mut hasher = std::collections::hash_map::DefaultHasher::new();
+            transfer.hash(&mut hasher);
+            hasher.finish()
+        })
```

Comment:
Instead of hashing all transfers together (which looses all valuable information), I think returning a hash of each would make more sense. 
Implementing the Hash trait for `Transfers` and using it in our logging (when we debug print them) would make a good enhancement!

Reply:
This is doing them seperately no? This maps each transfer to its hash?

We could/should just add this on Transfers though, good point.